### PR TITLE
Gloo Fed Multicluster RBAC guide

### PR DESCRIPTION
**Pull request by ned1313**
# Description

Docs update for Gloo Federation. 

* Adding in changes suggested in PR #3387 
* Adding guide for Multicluster RBAC
* Adding guide for Read-only Console

### CODE REVIEW

File: docs/content/guides/gloo_federation/multicluster_rbac.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
+---
+title: Multicluster RBAC
+description: Applying role-based access control to multiple Gloo instances
+weight: 40
+---
+
+Gloo Federation allows you to administer multiple instances of Gloo across multiple Kubernetes clusters. Multicluster role-based access control is a feature of Gloo Federation that controls access and actions on Gloo instances across multiple Kubernetes clusters. The feature ensures that users only have access to the resources they are supposed to, no matter where they are, and which Gloo instance they belong to.
```
Comment (joekelley) [inline docs/content/guides/gloo_federation/multicluster_rbac.md:7]:
```suggestion
Gloo Federation allows you to administer multiple instances of Gloo across multiple Kubernetes clusters. One Gloo Federation object might modify configuration across many instances of Gloo across many Kubernetes clusters. Multicluster role-based access control is a feature of Gloo Federation that controls access and actions on Gloo Federation APIs that might reconfigure many Gloo instances. The feature ensures that users are only allowed to modify Gloo Federation resources that configure Gloo resources in clusters and namespaces that they have explicitly been granted access to in order to facilitate multitenancy in the Gloo Federation control plane.
```

Comment (joekelley) [reply docs/content/guides/gloo_federation/multicluster_rbac.md:7]: or something to this effect -- trying to highlight the ways that this goes above and beyond kubernetes rbac, and call out why it's useful in the context of GF

File: docs/content/guides/gloo_federation/multicluster_rbac.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+glooctl install federation --values values.yaml --license-key LICENSE_KEY
+```
+
+On an existing installation, the `enableMultiClusterRbac` setting can be updated by running `helm upgrade` and overriding the value. Enabling the Multicluster RBAC feature creates an RBAC webhook and pod enforcing permissions on the Gloo Federation API groups.
+
+In the demonstration environment, we will upgrade the existing Gloo Federation installation by running the following:
+
+```bash
+# Add the gloo-fed helm repo
+helm repo add gloo-fed https://storage.googleapis.com/gloo-fed-helm
```
Comment (joekelley) [inline docs/content/guides/gloo_federation/multicluster_rbac.md:38]: would you mind adding the helm option to the install doc as well?

File: docs/content/guides/gloo_federation/multicluster_rbac.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+# Update your repos 
+helm repo update
+
+# Upgrade your Gloo Federation deployment
+# Assumes your install is called gloo-fed in the gloo-fed namspace
+helm upgrade -n gloo-fed gloo-fed gloo-fed/gloo-fed --set enableMultiClusterRbac=true
```
Comment (joekelley) [inline docs/content/guides/gloo_federation/multicluster_rbac.md:45]: note to self: we will need to address the license key upgrade UX issue before sending this doc live

File: docs/content/guides/gloo_federation/multicluster_rbac.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+      - '*'
+  - apiGroup: fed.enterprise.gloo.solo.io
+    placements:
+    - clusters:
+      - '*'
+      namespaces:
+      - '*'
+```
+
+The `gloo-fed` MultiClusterRoleBinding associates the MultiClusterRole with the `gloo-fed` service account. Without the binding, the gloo-fed pod wouldn't be able to make changes to the Gloo Federation installation.
```
Comment (joekelley) [inline docs/content/guides/gloo_federation/multicluster_rbac.md:115]:
```suggestion
The `gloo-fed` MultiClusterRoleBinding associates the MultiClusterRole with the `gloo-fed` service account. Without the binding, the gloo-fed pod wouldn't be able to update the status of Gloo Federation API objects.
```

File: docs/content/guides/gloo_federation/multicluster_rbac.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+  subjects:
+  - kind: User
+    name: system:serviceaccount:gloo-fed:gloo-fed
+```
+
+The `gloo-fed-console` MultiClusterRole and MultiClusterRoleBinding grant the same set of permissions to the `gloo-fed-console` service account.
+
+## Create Roles and Bindings
+
+In the previous section we installed Multicluster RBAC. The process created a MultiClusterRole and MultiClusterRoleBinding for both the `gloo-fed` and `gloo-fed-console` service account. It did not create any role or binding for the kubernetes-admin account, which means that we cannot make any changes to the Gloo Federation custom resources.
```
Comment (joekelley) [inline docs/content/guides/gloo_federation/multicluster_rbac.md:131]:
```suggestion
In the previous section we installed Multicluster RBAC. The process created a MultiClusterRole and MultiClusterRoleBinding for both the `gloo-fed` and `gloo-fed-console` service account. It did not create any role or binding for the default kind user account, kubernetes-admin, which means that we cannot make any changes to the Gloo Federation custom resources.
```

File: docs/content/guides/gloo_federation/read_only_console.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+For the purposes of this example, we have two clusters `local` and `remote`. The local cluster is also running Gloo Federation in addition to Gloo Enterprise. The kubectl context for the local cluster is `kind-local` and the remote cluster is `kind-remote`.
+
+## Configure access to the console
+
+The Gloo Federation console is exposed by the `gloo-fed-console` service running in the `gloo-fed` namespace. The console is available on port 8090. In a production scenario, you could choose to create a new service exposing the console on an IP address available outside the cluster. For our example, we are going to port-forward the service to the local IP address of the machine running the demonstration environment.
+
+```bash
+# Get the local IP address
+LOCAL_IP=$(ip a | awk '/inet / { print $2 }' | sed -n 2p | cut -d "/" -f1)
```
Comment (joekelley) [inline docs/content/guides/gloo_federation/read_only_console.md:28]: This is a linux-only step, mac users should be able to just use localhost

Comment (ned1313) [reply docs/content/guides/gloo_federation/read_only_console.md:28]: Updated the command, this was specific to my setup using a virtual machine without a desktop.

File: docs/content/guides/gloo_federation/multicluster_rbac.md (hunk 2 above)
Comment (ned1313) [reply docs/content/guides/gloo_federation/multicluster_rbac.md:38]: Added!

**Pull request merged**